### PR TITLE
PipelineController Changes for Cassandra Bulk

### PR DESCRIPTION
**Pull request by VardhanThigle**
Refactoring changes in PipelineController to accommodate Cassandra Source.

### CODE REVIEW

File: v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/templates/DbConfigContainer.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+import com.google.cloud.teleport.v2.source.reader.io.IoWrapper;
+import com.google.cloud.teleport.v2.spanner.migrations.schema.ISchemaMapper;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.apache.beam.sdk.transforms.Wait;
+
+/** Container for DB Configuration. */
+public interface DbConfigContainer {
```
Comment (VardhanThigle) [inline v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/templates/DbConfigContainer.java:26]: @bharadwaj-aditya 
Thanks.
As you had suggested in the review sync, I have kept the interface free of inheritance.
(This definitely improves the code cleanliness here)

File: v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/templates/PipelineController.java [hunk 2]
```diff
@@ -218,16 +237,6 @@ private static String generateSuffix(String shardId, String tableName) {
     return suffix;
   }
 
-  @VisibleForTesting
-  static SpannerConfig createSpannerConfig(SourceDbToSpannerOptions options) {
```
Comment (VardhanThigle) [inline v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/templates/PipelineController.java:222]: This function was not referenced in the code. Removed as this was marked unused.

File: v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/templates/DbConfigContainer.java (hunk 1 above)
Comment (bharadwaj-aditya) [reply v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/templates/DbConfigContainer.java:26]: Can you elaborate on the java docs here.

The purpose of this class is to hold the relevant config to invoke migration of a given shard / Table. May make sense to clarify.

File: v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/templates/SourceDbToSpanner.java [hunk 3]
```diff
@@ -102,13 +102,18 @@ static PipelineResult run(SourceDbToSpannerOptions options) {
     SpannerConfig spannerConfig = createSpannerConfig(options);
 
     // Decide type and source of migration
-    if (options.getSourceConfigURL().startsWith("gs://")) {
+    // TODO(vardhanvthigle): Move this within pipelineController.
+    if (options.getSourceDbDialect().equals(SourceDbToSpannerOptions.CASSANDRA_SOURCE_DIALECT)) {
+      return PipelineController.executeCassandraMigration(options, pipeline, spannerConfig);
+    } else if (options.getSourceConfigURL().startsWith("gs://")) {
```
Comment (bharadwaj-aditya) [inline v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/templates/SourceDbToSpanner.java:108]: suggestion - have a switch case for sources - and have this logic within the JDBC/mysql/pg source to make it clearer.

**Review by bharadwaj-aditya — APPROVED** [review]
Minor suggestions. LGTM overall.

File: v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/templates/DbConfigContainer.java (hunk 1 above)
Comment (VardhanThigle) [reply v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/templates/DbConfigContainer.java:29]: Added a note.

File: v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/templates/SourceDbToSpanner.java (hunk 3 above)
Comment (VardhanThigle) [reply v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/templates/SourceDbToSpanner.java]: Incorporated. Thanks.

**Review by bharadwaj-aditya — APPROVED** [review]
LGTM

**Pull request merged**